ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -60,6 +62,10 @@
  * will be accepted and all scheduled writes will be notified of the failure.
  */
 class NettyResponseChannel implements RestResponseChannel {
+  // add to this list if the connection needs to be closed on certain errors on GET, DELETE and HEAD.
+  // for a POST, we always close the connection on error because we expect the channel to be in a bad state.
+  static final List<HttpResponseStatus> CLOSE_CONNECTION_ERROR_STATUSES = new ArrayList<>();
```

Comment:
So for now, for non-POST, connection will be kept alive regardless of the error type? Is that because the responses are fully formed even on errors? Also, how does the ambry client deal with keepalive today? (I would imagine it would be the other way around for it).